redbot/core/settings_caches.py

```diff
@@ -147,22 +147,24 @@ def __init__(self, config: Config):
             isinstance(r_or_u, str) for r_or_u in role_or_user
         ):
             raise TypeError("Whitelisted objects must be a list of ints")
+        if gid not in self._cached_whitelist:
+            self._cached_whitelist[gid] = []
```

Comment:
This `if` may result in adding empty list to the cache when in fact there is something in current whitelist which just wasn't cached just yet.